XIVComboPlugin/IconReplacer.cs

```diff
@@ -904,6 +916,116 @@ private ulong GetIconDetour(byte self, uint actionID)
                     return actionID;
                 }
             }
+            
+             //Pictomancer
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoSubtractivePallet))
+            {
+                if (actionID == PCT.Fire1)
+                {
+                    if (SearchBuffArray(PCT.SubPallet))
+                        return iconHook.Original(self, PCT.Bliz1);
+                    return iconHook.Original(self, PCT.Fire1);
+                }
+
+                if (actionID == PCT.Fire2)
+                {
+                    if (SearchBuffArray(PCT.SubPallet))
+                        return iconHook.Original(self, PCT.Bliz2);
+                    return iconHook.Original(self, PCT.Fire2);
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoHolyWhiteCombo))
+            {
+                if (actionID == PCT.HolyWhite)
+                {
+                    if (SearchBuffArray(PCT.Monochrome))
+                        return PCT.CometBlack;
+                    return PCT.HolyWhite;
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoMotifMuseFeature))
+            {
+                if (actionID == PCT.CreatureMotif)
+                {
+                    var PCTGauge = JobGauges.Get<PCTGauge>();
+                    if (PCTGauge.CreatureMotifDrawn)
+                        return iconHook.Original(self, PCT.LivingMuse);
+                    return iconHook.Original(self, actionID);
+                }
+
+                if (actionID == PCT.WeaponMotif)
+                {
+                    var PCTGauge = JobGauges.Get<PCTGauge>();
+                    if (PCTGauge.WeaponMotifDrawn)
+                        return iconHook.Original(self, PCT.SteelMuse);
+                    if(SearchBuffArray(PCT.HammerReady))
+                        return iconHook.Original(self, PCT.HammerStamp);
+                    return iconHook.Original(self, actionID);
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoStarrySkyCombo))
+            {
+                if (actionID == PCT.LandscapeMotif)
+                {
+                    var PCTGauge = JobGauges.Get<PCTGauge>();
+                    if (PCTGauge.LandscapeMotifDrawn)
+                        return PCT.StarryMuse;
+                    if (SearchBuffArray(PCT.StarStruck))
+                        return PCT.StarPrism;
+                    return PCT.StarryMotif;
+                }
+            }
+            
+            //VIPER
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperDeathRattleMaximum))
+            {
+                if (actionID == VPR.SteelFangs || actionID == VPR.DreadFangs)
+                {
+                    if (iconHook.Original(self, VPR.SerpentsTail) == VPR.DeathRattle)
```

Comment:
I prefer to only call `Original` for the return value:

```suggestion
                    switch(lastMove)
                    {
                        case VPR.FlankStrike:
                        case VPR.FlankFang:
                        case VPR.HindStrike:
                        case VPR.HindFang:
                            return VPR.DeathRattle;
                        default:
                            return iconHook.Original(self, actionID);
                    }
```

Reply:
Unfortunately this creates an issue where the icon doesn't revert to Steel Fang/Dread Fang after using DeathRattle, I guess because lastMove isn't counting DeathRattle as the most recent action.

Also, lastMove is a signed int but we're comparing against unsigned ints, so the switch statement won't work as written. We either need to cast to uint or change lastMove to be ReadInt64 instead of ReadInt32. I'm not sure what other fallout that would have with other comparisons